File: Majority_Element/submission2.py
class Solution(object):
    def majorityElement(self, nums):
        
        count = 1
        num = nums[0]
        n = len(nums)

        for i in range(1, n):
            if nums[i]==num:
                count += 1
            else:
                count -=1
                if count==0:
                    num = nums[i]
                    count = 1
        
        return num

File: Find_Minimum_in_Rotated_Sorted_Array/submission2.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int n = nums.size(), s = 0, e = n-1, mid;
        while (s<e){
            mid = s + (e-s)/2;
            if (nums[mid] > nums[e]){
                s = mid + 1;
            } else {
                e = mid;
            }
        }

        return nums[s];
    }
};